### PR TITLE
Update to Angular 13

### LINKED ISSUE #20: Upgrade to Angular 13
See [the relevant PR in Lab 3](https://github.com/UMM-CSci-3601/3601-lab3_angular-javalin/pull/741) for info on how that played out.

### PR DESCRIPTION
**Pull request by NicMcPhee**
Upgrade from Angular 11 to Angular 13 following the [Angular update guide](https://update.angular.io/?l=2&v=11.0-13.0) and notes from [doing this on Lab 3](https://github.com/UMM-CSci-3601/3601-lab3_angular-javalin/pull/741).

As noted by @floogulinc on Slack, it's useful to update `flex-layout` at the same time as `material` to avoid peer dependency issues. So, for example, the second step in the migration guide becomes:

```text
npx @angular/cli@12 update @angular/material@12 @angular/flex-layout@12.0.0-beta.35 -C
```

I also [updated the Karma coverage tooling](https://mrjean.be/posts/update-karma-coverage-reporting-for-use-in-angular-11/) while I was here.

### CODE REVIEW

File: client/src/test.ts [hunk 1]
```diff
@@ -9,17 +9,19 @@ import {
...
   };
 };
 
 // First, initialize the Angular testing environment.
 getTestBed().initTestEnvironment(
   BrowserDynamicTestingModule,
-  platformBrowserDynamicTesting()
+  platformBrowserDynamicTesting(), {
+    teardown: { destroyAfterEach: false }
+}
```
Comment (helloworld12321) [inline client/src/test.ts]: Quick indent fix

```suggestion
  platformBrowserDynamicTesting(),
  {
    teardown: { destroyAfterEach: false }
  }
```

**Review by helloworld12321 — DISMISSED** [review]
Looks good, thanks so much! 🙂

(If you commit that indentation fix, I can re-review.)

File: client/src/test.ts (hunk 1 above)
Comment (NicMcPhee) [reply client/src/test.ts]: Thanks!

**Review by helloworld12321 — APPROVED** [review]
👍✅

**Pull request merged**